src/Checkbox.js

```diff
@@ -0,0 +1,195 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import { rem } from 'polished';
+
+import Box from './Box';
+import Flex from './Flex';
+import Text from './Text';
+import Icon from './Icon';
+
+const CheckIcon = ({ fill = 'currentColor', ...props }) => (
+  <svg width="10" height="7" viewBox="0 0 10 7" {...props}>
+    <path
+      fill={fill}
+      fillRule="evenodd"
+      d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
+    />
+  </svg>
+);
+
+CheckIcon.propTypes = {
+  fill: PropTypes.string.isRequired
+};
+
+const MinusIcon = ({ fill = 'currentColor', ...props }) => (
+  <svg width="16" height="16" viewBox="0 0 16 16" {...props}>
+    <path
+      fill={fill}
+      fillRule="evenodd"
+      d="M11 7H5c-.55 0-1 .45-1 1s.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1z"
+    />
+  </svg>
+);
+
+MinusIcon.propTypes = {
+  fill: PropTypes.string.isRequired
+};
+
+const Label = styled(Flex.withComponent('label'))`
+  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
+`;
+
+const Input = styled(Box.withComponent('input'))`
+  clip: rect(1px, 1px, 1px, 1px);
+  height: 1px;
+  position: absolute;
+  white-space: nowrap;
+  width: 1px;
+  opacity: 0;
+  overflow: hidden;
+  padding: 0;
+`;
+
+const StyledCheckbox = ({ children, ...props }) => (
+  <Flex
+    bg="white"
+    border="1px solid"
+    borderColor="grey60"
+    borderRadius="small"
+    color="white"
+    alignItems="center"
+    justifyContent="center"
+    width={rem('16px')}
+    height={rem('16px')}
+    {...props}
+  >
+    {children}
+  </Flex>
+);
+
+const Checkbox = ({
+  id,
+  name,
+  label,
+  disabled,
+  isInvalid,
+  checked,
+  onChange,
+  value,
+  indeterminate,
+  icon,
+  ...props
+}) => (
+  <Label alignItems="center" my="regular" {...props}>
```

Comment:
It feels a little confusing that the label prop is optional, yet this always exists.

Reply:
A label always needs to be provided in the code for good accessibility, but it can be visually hidden by leaving out the `label` prop. `Label` wraps the hidden html checkbox, the styled checkbox (whatever we choose to name it), and the label - associating them implicitly and allowing us to style the checkbox and/or hide the label text as we please, while still supporting assistive technologies.  https://www.w3.org/WAI/tutorials/forms/labels/#hiding-label-text